### PR TITLE
Fix connection leak

### LINKED ISSUE #916: 2.0: API - Ulimit on boxes needs to be set during provisioning.
**Short Summary:**
Ulimit on boxes need to be set on boxes while provisioning
**Steps to Reproduce:**
n/a
**Expected Results:**
n/a
**Actual Results:**
n/a
The ulimit for open files looks to be set at 1024, this needs to be increased to something high

### PR DESCRIPTION
**Pull request by stgleb**
Reuse client and transport  objects to prevent connection leak between SG and kube-api
Bu,p k8s version to include https://github.com/kubernetes/kubernetes/pull/68041

Closes #916

### CODE REVIEW

File: pkg/controlplane/server.go [hunk 1]
```diff
@@ -286,6 +288,11 @@ func configureApplication(cfg *Config) (*mux.Router, error) {
 	}
 	protectedAPI.Use(authMiddleware.AuthMiddleware, api.ContentTypeJSON)
 
+	// Run pprof http server
+	go func(){
+		logrus.Info(http.ListenAndServe("localhost:6060", nil))
```
Comment (saheienko) [inline pkg/controlplane/server.go]: Please, open a separate PR for this feature.
Also, it shouldn't be enabled by default.

Comment (egopher) [reply pkg/controlplane/server.go]: Can it be optional with a flag?

File: pkg/kube/handler.go [hunk 2]
```diff
@@ -916,7 +986,7 @@ func (h *Handler) getClusterMetrics(w http.ResponseWriter, r *http.Request) {
 
 	for metricType, relUrl := range metricsRelUrls {
 		url := fmt.Sprintf("https://%s/%s/%s", masterNode.PublicIp, baseUrl, relUrl)
-		metricResponse, err := h.getMetrics(url)
+		metricResponse, err := h.getMetrics(url, k)
```
Comment (saheienko) [inline pkg/kube/handler.go]: Could you use kubernetes client for retrieving these metrics? Kubernetes client solves this issues.

Comment (stgleb) [reply pkg/kube/handler.go]: Will do in separate PR, just want to fix it as fast as possible.

**Pull request merged**